### PR TITLE
Lock the repo during calls to git

### PR DESCRIPTION
**Pull request by Wuestengecko**
This fixes an issue where the invoked git process would return an error if a parallel process was already performing some task within the repo (like fetching updates). Instead, we now use an additional `capellambse.lock` file within the repo directory which we can `flock` to make sure that only one git instance will be called at a time.

This `flock` based approach works regardless of which parallelization method is used, i.e. thread-based and process-based both work fine.

### CODE REVIEW

File: capellambse/helpers.py [hunk 1]
```diff
@@ -116,6 +127,50 @@ def normalize_pure_path(
...
+
+else:
+
+    @contextlib.contextmanager
+    def flock(file: pathlib.Path) -> t.Iterator[None]:
+        file = file.resolve()
+        with file.open("wb") as lock:
+            try:
+                fcntl.flock(lock, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            except OSError as err:
```
Comment (MoritzWeber0) [inline capellambse/helpers.py:164]: Can't we just use `BlockingIOError` here? `errno.EAGAIN` should always fall under `BlockingIOError` and all other errors are raised anyway.

**Review by MoritzWeber0 — COMMENTED** [review]
Only one short comment, I'm not ready with the review yet. Most likely, I will not have time before Monday for a complete review, sorry.

File: capellambse/helpers.py (hunk 1 above)
Comment (Wuestengecko) [reply capellambse/helpers.py:165]: You're right, good catch!

File: tests/test_helpers.py [hunk 2]
```diff
@@ -93,3 +97,72 @@ def test_flatten_html_formats_unordered_lists(
     input: str, expected: str
 ) -> None:
     assert helpers.flatten_html_string(input) == expected
+
+
+if sys.platform.startswith("win"):
+    import msvcrt
+
```
Comment (ewuerger) [inline tests/test_helpers.py:104]: To spare the indentation one can use `@pytest.mark.skipif(sys.platform.startswith("linux"), reason="does not run on linux")`. The other test could use `@pytest.mark.skipif(sys.platform.startswith("win"), reason="does not run on windows")`. The import would happen in the body of the test function.

**Review by ewuerger — CHANGES_REQUESTED** [review]
Looks good. LGTM.

**Pull request merged**